src/approximations/nystrom.jl

```diff
@@ -9,9 +9,9 @@ function sampleindex(X::AbstractVector, r::Real)
     return S
 end
 
-@deprecate sampleindex(X::AbstractMatrix, r::Real; obsdim::Integer=defaultobs) sampleindex(
-    vec_of_vecs(X; obsdim=obsdim), r
-) false
+@deprecate sampleindex(
+    X::AbstractMatrix, r::Real; obsdim::Union{Integer,Nothing}=defaultobs
+) sampleindex(vec_of_vecs(X; obsdim=obsdim === nothing ? nothing : Int(obsdim)), r) false
```

Comment:
What's the benefit of the ternary ? : over just
```suggestion
) sampleindex(vec_of_vecs(X; obsdim=obsdim), r) false
```
?

Reply:
is it because here it was `Integer` whereas `vec_of_vecs` requires more specific `Int`?

Reply:
Yes, exactly. The current deprecation is not correct since it does not handle non-Int Integers. It doesn't matter much since we're allowed to remove it any time but since I had to change it anyway I fixed it.